src/Core/Managed/Shared/DataContracts/DependencyTelemetry.cs

```diff
@@ -201,6 +214,7 @@ void ITelemetry.Sanitize()
         {
             this.Name = this.Name.SanitizeName();
             this.Name = Utils.PopulateRequiredStringValue(this.Name, "name", typeof(DependencyTelemetry).FullName);
+            this.Id = this.Id.SanitizeName();
```

Comment:
Id needs to be initialized in the constructor the same way it is done for RequestTelemetry.Id. ResultCode needs to be Sanitized.